RadixWallet/Clients/FactorSourcesClient/FactorSourcesClient+Interface.swift

```diff
@@ -173,7 +185,26 @@ public struct SigningFactor: Sendable, Hashable, Identifiable {
 	}
 }
 
+extension DeviceFactorSource {
+	func removingMainFlag() -> Self {
```

Comment:
why not mutating?

Reply:
because the `updateFactorSource` does not work with mutating FactorSource closure (inout), but rather takes the already changed factorSource, see usage below:

```swift
for oldMainBDFS in oldMainBDFSSources {
	try await updateFactorSource(oldMainBDFS.removingMainFlag().embed())
}
```

Reply:
having `for var oldMainBDFS` will not work?
anyway, this is minor improvement.

Reply:
Sure but that would result in this code which is less pretty IMO:

```swift
for var oldMainBDFS in oldMainBDFSSources {
	oldMainBDFS.removeMainFlag()
	try await updateFactorSource(oldMainBDFS.embed())
}
```

Reply:
Can't it be still written like this:
```swift
for var oldMainBDFS in oldMainBDFSSources {
	try await updateFactorSource(oldMainBDFS.removeMainFlag().embed())
}
```
asking just out of curiosity.

Reply:
no, not if it does not return `Self` right? Otherwise we would be calling `embed()` on `Void`..? An a function which modifies self and returns self, can just as well not modify self, but rather a copy, thus not having to be marked `mutating`.

Also you just proposed:

```swift
for var oldMainBDFS in oldMainBDFSSources {
	try await updateFactorSource(oldMainBDFS.removeMainFlag().embed())
}
```

which is is worse than current impl? :) 

```swift
for oldMainBDFS in oldMainBDFSSources {
	try await updateFactorSource(oldMainBDFS.removingMainFlag().embed())
}
```

since you have just added one word `var`, without gaining anything really? :) 

I'm following the standard Swift naming convention which is to use `ing` for methods which are not `mutating` and which are returning `Self` (a copy of). e.g.:

```swift
trimming
partitioningIndex
replacing
applying
```

Reply:
Fair enough, but mutating functions in their essence do exactly what you did implement `removingMainFlag`, you can say that mutating is a syntactic sugar for your implementation.

Anyway, this is a matter of taste, no point in spending time on this.